### PR TITLE
compiler/next: Stop using public fields in scope-types

### PR DESCRIPTION
**Pull request by aconsroe-hpe**
* Makes public fields private behind getter methods where appropriate.
* Adds an `Iterable` template as a more opaque "thing to be iterated over"
* Adds doc comments to classes, getters, and enums

### CODE REVIEW

File: compiler/next/include/chpl/resolution/scope-types.h [hunk 1]
```diff
@@ -248,63 +251,120 @@ class Scope {
...
 // could fold into the logic about variable declarations).
-struct ResolvedVisibilityScope {
-  const Scope* scope;
-  std::vector<VisibilitySymbols> visibilityClauses;
+class ResolvedVisibilityScope {
+private:
+  const Scope* scope_;
+  std::vector<VisibilitySymbols> visibilityClauses_;
+
+public:
```
Comment (mppf) [inline compiler/next/include/chpl/resolution/scope-types.h:344]: Do you think we should
 a) allow a mix of indentation style for public/private in class bodies
 b) never indent them (like here)
 c) indent them with one space (like other compiler/next code I've been writing)
?

I am curious what you think about it but it should not hold up this PR.

**Review by mppf — APPROVED** [review]
Looks great, thanks for working through these

**Comment by mppf** [comment]
Looks like there are some CI failures:

> /home/runner/work/chapel/chapel/doc/rst/developer/compiler-internals/resolution.rst:11: WARNING: cpp:identifier reference target not found: Iterable<decltype(visibilityClauses_)>
/home/runner/work/chapel/chapel/doc/rst/developer/compiler-internals/resolution.rst:11: WARNING: cpp:identifier reference target not found: visibilityClauses_

Please see https://github.com/chapel-lang/chapel/blob/main/compiler/next/doc/doxygen-pitfalls.txt and if it's an issue not covered there we should add it.

**Comment by aconsroe-hpe** [comment]
Looking into the docs thing

File: compiler/next/include/chpl/resolution/scope-types.h (hunk 1 above)
Comment (aconsroe-hpe) [reply compiler/next/include/chpl/resolution/scope-types.h]: I'd rather have one style and I've adjusted to match the existing one. We could look at using `clang-format` everywhere to all be on the same page. The right configuration var is `AccessModifierOffset -1`

**Comment by aconsroe-hpe** [comment]
Even with
```diff
diff --git a/compiler/next/include/chpl/resolution/scope-types.h b/compiler/next/include/chpl/resolution/scope-types.h
index 55debbe3f6..be59bcdc49 100644
--- a/compiler/next/include/chpl/resolution/scope-types.h
+++ b/compiler/next/include/chpl/resolution/scope-types.h
@@ -342,7 +342,7 @@ class ResolvedVisibilityScope {
   std::vector<VisibilitySymbols> visibilityClauses_;
 
  public:
-  using VisibilitySymbolsIterable = Iterable<decltype(visibilityClauses_)>;
+  using VisibilitySymbolsIterable = Iterable<std::vector<VisibilitySymbols>>;
 
   ResolvedVisibilityScope(const Scope* scope)
     : scope_(scope)
```

I still get `/home/andrew/Repos/chapel/doc/rst/developer/compiler-internals/resolution.rst:11: WARNING: cpp:identifier reference target not found: Iterable<std::vector<VisibilitySymbols>>`, so is doxygen not picking up the `iteration.h`? Or it can't follow past a `using`?

File: compiler/next/include/chpl/resolution/scope-types.h (hunk 1 above)
Comment (mppf) [reply compiler/next/include/chpl/resolution/scope-types.h]: FWIW I think that clang-format would be a great practice for us to get in to but I havn't looked at using it myself. I don't have any idea of the difficulty / level of effort to using it.

**Comment by mppf** [comment]
I don't know what the doxygen issue but you can try removing the `using` and see if that fixes it.

**Comment by aconsroe-hpe** [comment]
For future reference, we resolved the doxygen issue by adding `chpl::Iterable` to `base.rst` so it would have a target for the link. The `using VisibilitySymbolsIterable` inside a class was not an issue. The `using It = typename C::const_iterator` was an issue and fixed by just replacing with the rhs.

**Pull request merged**